packages/components/src/tree-grid/roving-tab-index-item.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useRovingTabIndexContext } from './roving-tab-index-context';
+import type { RovingTabIndexItemProps } from './types';
+
+export const RovingTabIndexItem = forwardRef(
+	function UnforwardedRovingTabIndexItem(
+		{ children, as: Component, ...props }: RovingTabIndexItemProps,
+		forwardedRef: React.ForwardedRef< any >
+	) {
+		const localRef = useRef< any >();
+		const ref = forwardedRef || localRef;
+		// @ts-expect-error - We actually want to throw an error if this is undefined.
+		const { lastFocusedElement, setLastFocusedElement } =
+			useRovingTabIndexContext();
+		let tabIndex;
+
+		if ( lastFocusedElement ) {
+			tabIndex =
+				lastFocusedElement ===
+				// TODO: The original implementation simply used `ref.current` here, assuming
+				// that a forwarded ref would always be an object, which is not necessarily true.
+				// This workaround maintains the original runtime behavior in a type-safe way,
+				// but should be revisited.
+				( 'current' in ref ? ref.current : undefined )
+					? 0
+					: -1;
+		}
+
+		const onFocus: React.FocusEventHandler< Element > = ( event ) =>
+			setLastFocusedElement?.( event.target );
+		const allProps = { ref, tabIndex, onFocus, ...props };
+
+		if ( typeof children === 'function' ) {
+			return children( allProps );
+		}
+
+		if ( ! Component ) return null;
```

Comment:
Just an added guard, but nevertheless a runtime change.

Interestingly, there are no actual usages of this `as` prop in the repo.

Reply:
Did some more investigation — this component was introduced as a separate component from `TreeGrid` and it seemed to have the `as` prop from its beginnings: https://github.com/WordPress/gutenberg/pull/18014/files#diff-b3f07d9e77626753ea74b8a24a73ef0f82bc0a9d537617976b8ad34415f13ac0

The `as` prop listed in the docs, in the Storybook examples, and even in the unit tests, but I don't think other components were using it at that point.

The component was later incorporated into `TreeGrid` as part of https://github.com/WordPress/gutenberg/pull/22373